channels/rail/client/rail_main.c

```diff
@@ -124,13 +126,18 @@ static UINT rail_client_execute(RailClientContext* context,
 {
 	char* exeOrFile;
 	UINT error;
-	railPlugin* rail = (railPlugin*) context->handle;
+	railPlugin* rail;
+
+	if (!context || !exec)
+		return ERROR_INVALID_PARAMETER;
+
+	rail = (railPlugin*) context->handle;
 	exeOrFile = exec->RemoteApplicationProgram;
 
 	if (!exeOrFile)
 		return ERROR_INVALID_PARAMETER;
 
-	if (strlen(exeOrFile) >= 2)
+	if (strnlen(exeOrFile, MAX_PATH) >= 2)
 	{
```

Comment:
In this file it's the only pertinent change, all others are unneeded checks. 
Verify the call stacks and you'll see that the rail argument can never be NULL (or it would have failed a long before) and the same apply for the second parameter of each function (called with a ref on local stack variable).